schoolopy/authentication.py

```diff
@@ -0,0 +1,135 @@
+import random
+import time
+import yaml
+import requests_oauthlib
+import json
+from urllib import parse
+import requests
+
+class SchoologyAuth:
```

Comment:
I don't know if auth or authentication is the best thing to call this, nor if this is the best method of implementation. I think a good solution would be to merge this into the main `Schoology` class, and allow the number of auth legs to be specified when instantiating the class.